server/src/repositories/asset.repository.ts

```diff
@@ -709,4 +713,27 @@ export class AssetRepository implements IAssetRepository {
   async upsertFiles(files: { assetId: string; type: AssetFileType; path: string }[]): Promise<void> {
     await this.fileRepository.upsert(files, { conflictPaths: ['assetId', 'type'] });
   }
+
+  updateOffline(importPaths: string[], exclusionPatterns: string[]): Promise<UpdateResult> {
+    const paths = importPaths.map((importPath) => `${importPath}%`).join('|');
+    const exclusions = exclusionPatterns.map((pattern) => globToSqlPattern(pattern)).join('|');
+    return this.repository
+      .createQueryBuilder()
+      .update()
+      .set({
+        isOffline: true,
+        deletedAt: new Date(),
```

Comment:
The `status` also needs to be set. This is why I don't really like the status field. The same info is stored in multiple places so it's so easy for it to go out of sync like this.

Reply:
I'm trying to remember how the status field works. We use it to distinguish between user-deleted and offlined items, right? What shuld it be set to here?

Reply:
> The `status` also needs to be set. This is why I don't really like the status field. The same info is stored in multiple places so it's so easy for it to go out of sync like this.

In main we don't set the status field here, if we want to change this we should do that in a separate PR.